### PR TITLE
Simplify flatteining and inlining of Coreschema

### PR DESCRIPTION
**Pull request by adriangb**
Selected Reviewer: @davidhewitt

### CODE REVIEW

**Comment by adriangb** [comment]
please review

File: tests/test_model_signature.py [hunk 1]
```diff
@@ -38,7 +38,11 @@ def test_generic_model_signature():
     class Model(BaseModel, Generic[T]):
         a: T
 
-    sig = signature(Model[int])
+    IntModel = Model[int]
+
+    assert IntModel.model_validate({'a': '1'}).a == 1
+
+    sig = signature(IntModel)
```
Comment (davidhewitt) [inline tests/test_model_signature.py:45]: It looks like the cause of this is that the following code is firing on this branch from `_model_construction.py`:

```
    if collect_invalid_schemas(schema):
        set_model_mocks(cls, cls_name)
        return False
```

I suggest we understand this further - smells like a possible bug.

Comment (davidhewitt) [reply tests/test_model_signature.py:45]: Let's xfail instead of modify, I will dig into this in parallel.

**Comment by JrooTJunior** [comment]
@dmontagu, @samuelcolvin, this changes breaks compatibility of https://github.com/aiogram/aiogram with pydantic 2.4

```
  File "/home/jrootjunior/work/aiogram/aiogram3/.hatch/dev/lib/python3.11/site-packages/pydantic/_internal/_core_utils.py", line 515, in count_refs
    recurse(state['definitions'][ref], count_refs)
            ~~~~~~~~~~~~~~~~~~~~^^^^^
KeyError: 'aiogram.types.input_text_message_content.InputTextMessageContent:94580207914144'
```

([full traceback](https://gist.github.com/JrooTJunior/7e3f6d3100184e6a1381eab8302eb94d))

You can check it by installing `pip install aiogram==3.1.0` and then just import it - `import aiogram`.
I have temporary workaround - restrict to install pydantic 2.4 with aiogram, so, when you install `pip install aiogram=3.1.1` and then import it - it works as well.

**Comment by davidhewitt** [comment]
Thanks for the report, we're investigating.

**Comment by Olegt0rr** [comment]
Just leave it here

Screenshot 2023-09-25 at 20 15 16

File: tests/test_root_model.py [hunk 2]
```diff
@@ -47,11 +47,6 @@ def check_schema(schema: CoreSchema) -> None:
     # we assume the shape of the core schema here, which is not a guarantee
     # pydantic makes to its users but is useful to check here to make sure
     # we are doing the right thing internally
```
Comment (lig) [inline tests/test_root_model.py:49]: Remove related comment as well then?

**Comment by adriangb** [comment]
> Just leave it here
> 
> Screenshot 2023-09-25 at 20 15 16

What was the intention of posting this? Do you feel this was an _intentional_ breaking change?

**Pull request merged**